intercept/src/intercept.cpp

```diff
@@ -9246,6 +9568,14 @@ void CLIntercept::dumpProgramBinary(
         {
             for( size_t i = 0; i < numDevices; i++ )
             {
+                CLIntercept* pIntercept = GetIntercept();
+                if (((pIntercept->config().DumpReplayKernelEnqueue != -1) || pIntercept->config().DumpReplayKernelName != "") && 
+                      !pIntercept->config().DumpProgramBinaries )
+                {
+                    auto& binaryVectorVector = m_DeviceBinaryMap[devices[i]];
+                    binaryVectorVector.push_back({programBinaries[i], programBinaries[i] + programBinarySizes[i]});
+                    return;
```

Comment:
Careful with the early return here - this will leak memory.

I'm also not sure if this is what is intended since this will only save the program binary for the first device.

Is there a way to restructure this code that doesn't need an early return?  Even better, could we query the program binaries when they are needed, and avoid saving them away completely?

Reply:
Yes to your last point. When I implemented this, I wasn't aware of how Windows messes up writing to a file with `std::ofstream` if you don't specify binary mode.

I will change it back, good catch also on the memory leak. I didn't see the manual allocations above!